pkg/asset/machines/aws/instance_types.go

```diff
@@ -38,17 +73,17 @@ func PreferredInstanceType(ctx context.Context, meta *awsconfig.Metadata, types
 		LocationType: aws.String("availability-zone"),
 	})
 	if err != nil {
-		return types[0], err
+		return found, err
 	}
-	reqZones := sets.NewString(zones...)
-	found := map[string][]string{}
+
+	// iterate through the offerings and create a new map of sets.String
```

Comment:
This comment is not quite accurate any more.

Reply:
changed.